README.md

```diff
@@ -18,33 +18,39 @@ ccache --max-size=50G
 Cloning repos
 -------------
 
-These can be skipped depending on your circumstances:
+### Guidelines:
 - linux-firmware: Not necessary if your distribution already contains firmware for your GPU. You can find your current firmware in `/lib/firmware/amdgpu`. The firmware is installed by copying files from the firmware repository into that directory and re-installing the kernel (which packs the firmware into /boot/initrd*). The kernel only loads firmware from initrd.
-- meson, libva, wayland-protocols (and the wayland dependency) are not needed if Mesa doesn't fail to configure. Ubuntu 20.04 needs them all. Ubuntu 22.04 might not.
-- libdrm can be skipped if Mesa doen't fail to configure, but that's rare.
+- meson, libva, wayland-protocols (and the wayland dependency) are not needed if Mesa doesn't fail to configure. On Ubuntu 22.04, these dependencies should be met if the above mentioned list of packages are installed. Otherwise, build and install them from source.
+- libdrm can be skipped if Mesa doesn't fail to configure, but that's rare.
 - xf86-video-amdgpu is almost always not needed unless somebody explicitly told you that you need it.
 - The 32-bit driver is not needed if Steam isn't going to be used because only Steam and some Steam games use it.
 
+The optional sources can be skipped depending on your circumstances. Use your discretion to pick and choose from the optional sources, if necessary.
 Clone with ssh for the repositories where you will want to push. The below commands only give you read-only access.
 
-```bash
+### Optional sources to install
 
+```bash
+# Frequently installed dependencies
 git clone git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git # Ideally use the AMD internal repository instead
+git clone https://gitlab.freedesktop.org/agd5f/linux.git -b amd-staging-drm-next # Ideally use the AMD internal repository instead
 git clone https://gitlab.freedesktop.org/xorg/driver/xf86-video-amdgpu.git
+git clone https://gitlab.freedesktop.org/mesa/demos.git # just for glxinfo and glxgears
 
-# Dependencies
-git clone https://github.com/marekolsak/marek-build.git
+# Other dependencies
 git clone https://github.com/mesonbuild/meson.git build-meson
 git clone https://github.com/intel/libva.git
 git clone https://gitlab.freedesktop.org/wayland/wayland.git
 git clone https://gitlab.freedesktop.org/wayland/wayland-protocols.git
+```
+
+### Essential sources to install
 
+```bash
 # For the driver:
-git clone https://gitlab.freedesktop.org/agd5f/linux.git -b amd-staging-drm-next # Ideally use the AMD internal repository instead
 git clone https://gitlab.freedesktop.org/mesa/drm.git
 git clone https://github.com/llvm/llvm-project.git
 git clone https://gitlab.freedesktop.org/mesa/mesa.git
-git clone https://gitlab.freedesktop.org/mesa/demos.git # just for glxinfo and glxgears
```

Comment:
We need this to get glxinfo32 and glxgears32, which is the only way to smoke test 32-bit drivers.